runtime/op/traverse/ztests/head.yaml

```diff
@@ -1,4 +1,4 @@
-zed: over a | head 2
+zed: over a | head 2+1
```

Comment:
Rather than change this test, whose primary intent is to cover the interaction between `over` and `head`, let's add `compiler/ztests/head.yaml` with a few `zc -C -s` tests (similar to `compiler/ztests/from-error.yaml`).

Test cases I think we should include:
1. `head 1`
2. `const x=1 head x + 1`
3. `head 1.`
4. `head "1"`
5. `head x`

(`zc -C -s` should succeed for the first two and fail for the last three.)